src/net.cpp

```diff
@@ -1705,31 +1702,37 @@ void static Discover(boost::thread_group& threadGroup)
 void StartNode(boost::thread_group& threadGroup, CScheduler& scheduler)
 {
     uiInterface.InitMessage(_("Loading addresses..."));
-    // Load addresses for peers.dat
+    // Load addresses from peers.dat
     int64_t nStart = GetTimeMillis();
     {
         CAddrDB adb;
         if (!adb.Read(addrman))
             LogPrintf("Invalid or missing peers.dat; recreating\n");
+        else
+            LogPrintf("Loaded %i addresses from peers.dat  %dms\n", addrman.size(), GetTimeMillis() - nStart);
     }
 
-    //try to read stored banlist
+    uiInterface.InitMessage(_("Loading banned addresses..."));
+    // Load addresses from banlist.dat
+    nStart = GetTimeMillis();
     CBanDB bandb;
     banmap_t banmap;
-    if (!bandb.Read(banmap))
+    if (!bandb.Read(banmap)) {
         LogPrintf("Invalid or missing banlist.dat; recreating\n");
+    } else {
+        CNode::SetBanned(banmap); // thread save setter
+        CNode::SetBannedSetDirty(false); // no need to write down, just read data
```

Comment:
Why is it's default false (`net.cppL448)?
I would recommend to call `CNode::SetBannedSetDirty(false);` when we start up a node. There is no case where the list can be "dirty" in this state.